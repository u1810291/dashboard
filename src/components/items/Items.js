/** @jsx jsx */
import styled from '@emotion/styled';
import { css } from '@emotion/core';

export const aligns = [
  'auto',
  'normal',
  'start',
  'end',
  'center',
  'stretch',
  'baseline',
  'first baseline',
  'last baseline',
];

export const justifies = [
  'start',
  'end',
  'center',
  'stretch',
  'normal',
  'space-around',
  'space-between',
  'space-evenly',
];

export const flows = ['column', 'row'];

export const gaps = [0, 1, 2, 3, 4];

const Items = styled.div(
  ({
    inline = false,
    align = 'start',
    flow = 'column',
    justifyContent = 'normal',
    templateColumns = 'none',
    templateRows = 'none',
    alignContent = 'normal',
    justifyItems = 'normal',
    autoRows = 'auto',
    autoColumns = 'auto',
    gap = 2,
  }) => css`
    display: ${inline ? 'inline-grid' : 'grid'};
    grid-auto-flow: ${flow};
    grid-gap: calc(0.5 * ${gap} * var(--mgi-spacing));
    grid-template-columns: ${templateColumns};
    grid-template-rows: ${templateRows};
    align-items: ${align};
    align-content: ${alignContent};
    justify-content: ${justifyContent};
    justify-items: ${justifyItems};
    grid-auto-rows: ${autoRows};
    grid-auto-columns: ${autoColumns};
    // border: 1px solid rgba(255, 0, 0, 0.2);
  `,
);

export default Items;
