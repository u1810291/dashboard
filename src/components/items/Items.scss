:local(.root) {
  display: grid;
  grid-gap: var(--mgi-spacing);
  grid-auto-flow: column;
  justify-content: flex-start;

  &:not(.inline) {
    display: grid;
  }

  &.inline {
    display: inline-grid;
  }

  &.align-top {
    align-items: flex-start;
  }

  &.align-bottom {
    align-items: flex-end;
  }

  &.align-center {
    align-items: center;
  }

  &.align-stretch {
    align-items: stretch;
  }

  &.smallGap {
    grid-gap: calc(0.5 * var(--mgi-spacing));
  }

  &.extraGap {
    grid-gap: calc(2 * var(--mgi-spacing));
  }
}
