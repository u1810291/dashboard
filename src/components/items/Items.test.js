import React from 'react'
import { shallow } from 'enzyme'
import Items from '.'

describe('Items', () => {
  test('default', () => {
    const component = shallow(
      <Items>
        <div />
        <div />
        <div />
      </Items>
    )
    expect(component).toMatchSnapshot()
  })

  test('inline', () => {
    const component = shallow(
      <Items inline>
        <div />
        <div />
        <div />
      </Items>
    )
    expect(component).toMatchSnapshot()
  })

  test('align top', () => {
    const component = shallow(
      <Items align="top">
        <div />
        <div />
        <div />
      </Items>
    )
    expect(component).toMatchSnapshot()
  })

  test('align bottom', () => {
    const component = shallow(
      <Items align="bottom">
        <div />
        <div />
        <div />
      </Items>
    )
    expect(component).toMatchSnapshot()
  })

  test('align center', () => {
    const component = shallow(
      <Items align="center">
        <div />
        <div />
        <div />
      </Items>
    )
    expect(component).toMatchSnapshot()
  })

  test('align stretch', () => {
    const component = shallow(
      <Items align="stretch">
        <div />
        <div />
        <div />
      </Items>
    )
    expect(component).toMatchSnapshot()
  })

  test('template', () => {
    const component = shallow(
      <Items template="3fr 1fr 3fr">
        <div />
        <div />
        <div />
      </Items>
    )
    expect(component).toMatchSnapshot()
  })
})
