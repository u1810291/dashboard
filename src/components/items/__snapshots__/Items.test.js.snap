// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items align bottom 1`] = `
<div
  className="align-bottom"
  style={
    Object {
      "gridTemplateColumns": "none",
    }
  }
>
  <div />
  <div />
  <div />
</div>
`;

exports[`Items align center 1`] = `
<div
  className="align-center"
  style={
    Object {
      "gridTemplateColumns": "none",
    }
  }
>
  <div />
  <div />
  <div />
</div>
`;

exports[`Items align stretch 1`] = `
<div
  className="align-stretch"
  style={
    Object {
      "gridTemplateColumns": "none",
    }
  }
>
  <div />
  <div />
  <div />
</div>
`;

exports[`Items align top 1`] = `
<div
  className="align-top"
  style={
    Object {
      "gridTemplateColumns": "none",
    }
  }
>
  <div />
  <div />
  <div />
</div>
`;

exports[`Items default 1`] = `
<div
  className="align-top"
  style={
    Object {
      "gridTemplateColumns": "none",
    }
  }
>
  <div />
  <div />
  <div />
</div>
`;

exports[`Items inline 1`] = `
<div
  className="align-top inline"
  style={
    Object {
      "gridTemplateColumns": "none",
    }
  }
>
  <div />
  <div />
  <div />
</div>
`;

exports[`Items template 1`] = `
<div
  className="align-top"
  style={
    Object {
      "gridTemplateColumns": "3fr 1fr 3fr",
    }
  }
>
  <div />
  <div />
  <div />
</div>
`;
